src/pool/poolClient.ts

```diff
@@ -362,6 +414,46 @@ export class Client {
   private getOrCreatePoolEvents() {
     return this.poolEvents
   }
+  private getOrCreateUserService(userAddress:string,tokenAddress:string){
```

Comment:
user service used to be stateless, but now queries and caches events from the token contract